docs/extensibility/proactive-strategy.md

```diff
@@ -0,0 +1,173 @@
+
+        if (stopwatch.Elapsed > _threshold)
+        {
+            // Create arguments that encapsulate the information about the event.
+            var args = new ThresholdExceededArguments(context, _threshold, stopwatch.Elapsed);
+
+            // Since we detected that this execution took longer than the threshold, we will report this as an resilience event.
+            _telemetry.Report(
+                new ResilienceEvent(ResilienceEventSeverity.Warning, "ExecutionThresholdExceeded"), // Pass the event severity and the event name
+                context, // Forward the context
+                 args); // Forward the arguments so any listeners can recognize this particular event
+
+            if (_thresholdExceeded is not null)
+            {
+                await _thresholdExceeded(args).ConfigureAwait(context.ContinueOnCapturedContext);
+            }
+        }
+
+        // Just return the outcome
+        return outcome;
+    }
+}
+```
+<!-- endSnippet -->
+
+Review the code and comments to understand the implementation. Take note of the `ThresholdExceededArguments` struct:
+
+<!-- snippet: ext-proactive-args -->
+```cs
+// Arguments-based structs encapsulate information about particular event that occurred inside resilience strategy.
+// They cna expose any properties that are relevant to the event.
+// For this event the actual duration of execution and the threshold that was exceeded are relevant.
+public readonly struct ThresholdExceededArguments
+{
+    public ThresholdExceededArguments(ResilienceContext context, TimeSpan threshold, TimeSpan duration)
+    {
+        Context = context;
+        Threshold = threshold;
+        Duration = duration;
+    }
+
+    public TimeSpan Threshold { get; }
+
+    public TimeSpan Duration { get; }
+
+    // By convention, all arguments should expose the "Context" property.
+    public ResilienceContext Context { get; }
+}
+```
+<!-- endSnippet -->
+
+Arguments should always have an `Arguments` suffix and include a `Context` property. Using arguments boosts the extensibility and maintainability of the API, as adding new members becomes a non-breaking change. The `ThresholdExceededArguments` provides details about the actual execution time and threshold, allowing listeners to respond to this event or supply a custom callback for such situations.
+
+## Options
+
+In the previous section, we implemented the `TimingResilienceStrategy`. Now, it's time to integrate it with Polly and its public API.
+
+Let's define the public `TimingResilienceStrategy` to configure our strategy:
+
+<!-- snippet: ext-proactive-options -->
+```cs
+public class TimingStrategyOptions : ResilienceStrategyOptions
+{
+    public TimingStrategyOptions()
+    {
+        // It's recommended to set the default name for the options so
+        // the consumer can get additional information in the telemetry.
```

Comment:
What additional information? If it's just the name I would probably say "the consumer can see it in the telemetry" if it allows access to more things, an example could be handy to help build a picture.

Reply:
Reworded, as these are not just for telemetry. They can be directly consumed.